openedx/core/djangoapps/content/course_structures/tasks.py

```diff
@@ -28,6 +29,11 @@ def _generate_course_structure(course_key):
                 "children": [unicode(child.scope_ids.usage_id) for child in children]
             }
 
+            if (curr_block.category == 'discussion' and
+                    hasattr(curr_block, 'discussion_id') and
+                    curr_block.discussion_id):
```

Comment:
Yes, during testing I ran into several blocks that had an empty string as their discussion id.